### DISCUSSION TITLE
firmware  v.0.21 correction when transmitting via PTT on an external speakerphone

### DISCUSSION THREAD

**Discussion by Polf1** [egzumer/uv-k5-firmware-custom#339] [General]
When connecting an external speakerphone in FM radio listening mode, pressing the PTT button of the speakerphone does not turn on the transmission. In order for the transmission to turn on after pressing the PTT button on the speakerphone, you must exit the FM radio listening mode. Can this be fixed? Thank you.

**Comment by ErikS-web** [comment]
_Title makes no sense. ?? !!_

The effect that **External PTT does NOT functions , as long a signal is received**, is depends on the hardware. It not only on FM-radio.

As is mentioned also in the manual, at [section External PTT](https://github.com/egzumer/uv-k5-firmware-custom/wiki/Button-functions#ptt-1)

But try this hardware modification within your microphone #267

![Add-elco-in-microphone](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/a9b58be5-df76-4761-8019-d30c01c071d3)

With 47uF I was not satisfied with the sound on low-frequenties, so I placed 100uF in the microphone (type Baofeng K-780MIC). Perhabs 220 is even better.

_I'm curious about your finding_

**Reply by Polf1** [reply]
Thank you. I understand that the external PTT does not work the same way as the internal one. I think this is wrong. This is originally inherent in the firmware from Quansheng. For other similar transceivers, for example, the chinese Linton LT6100+, the external PTT works like an internal one and it is convenient. Example: a spinningist in a boat listens to an FM radio, an external microphone is fixed in a convenient place, the transceiver is in his pocket, he can quickly switch to transmission at any time, otherwise he needs to take the transceiver out of his pocket and press the internal PTT or turn off FM radio reception. I thought that the Egzumer firmware could fix the flaw. 
As for the refinement of the external speakerphone, it is not relevant for me, since both transceivers use 1.4 boards. Solder bridge on the 1.4 board does not need to be removed, since the pad of the capacitor and the connector connected by a printed conductor.

**Comment by ErikS-web** [comment]
Just try adding the capacitor **inside the external microphone.** 
That is not only the solution for the DC-voltage on speaker,-line  (#267) but also for this/your problem as described here in this thread.

Don't make any internal modification inside the radio.
It is a less than 1$ solution and works here. And the modified mic. also works on other type of my radio's

Open the your Mic. and make a picture and I will guide you.

For testing purpose, I have wired the extra switch PTT-2 to short the extra placed capacitor, when PTT-2 is pressed.
So I can reproduce this error you mentioned.

p.s. This problem cannot be fixed in the firmware because it is a hardware issue on Rev 1.4 board.

**Reply by ErikS-web** [reply]
Ok I understand you mixed up the other cause.

I made an extra comment in #267.
That this specific (rev. 1.4) case is reintroduced on rev 1.6 when desoldering is done to solve the DC-voltage issue.
And the elco solves this issue also ;-)

**Reply by Polf1** [reply]
I added a capacitor in series with the speaker. Nothing has changed, the external PTT does not work when receiving the signal.
![DSCN0495](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/31a8fc5e-5889-4d92-a7cc-6a00e3d59587)
![DSCN0496](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/01fe1620-e697-4441-9f78-9976ac61953b)
![DSCN0497](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/e9b6aebb-d6fd-48f2-b37d-5766e9229cb2)

**Reply by ErikS-web** [reply]
I think it is because of the volume potentiometer of 100 Ohm
That is between SP+ and GND and the Mid-contact to the speaker (via a 2.5 mm insertplug)
Than there's a continue DC-current via the 100 Ohm.

In your case I would "cut the red-wire" and place the elco in the SPK+ signal.
Or lift the left Leg of the potentiometer and insert the elco there.

p.s. 
What kind of radio with rev 1.4 board do you have:  K5 (white screen) or K58 (orange screen) ?

**Reply by Polf1** [reply]
OK, I'll take a microphone without a regulator and try it. I have a UV-K5(8) transceiver with yellow backlight.

**Reply by Polf1** [reply]
Thanks!  You were right. Disconnected the red wire and soldered the capacitor, it works!
![DSCN0498](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/c17c540d-e1b7-4cb2-bf21-d086a219566d)
![DSCN0499](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/b5b2ad78-1213-44da-9ffe-a8c30c6919c5)
![DSCN0500](https://github.com/egzumer/uv-k5-firmware-custom/assets/64047332/bdf48b1f-472e-4e37-98a6-3e79cd970d3e)

**Comment by ErikS-web** [comment]
Glad to be of help.

Enjoy the radio + mic ;-)
